specs/CoreWebView2.CreateFromCOMObject.md

```diff
@@ -47,8 +53,52 @@ public class MyWebView2Control
 }
 ```
 
+## WinRT to COM
+
+```c++
+winrt::com_ptr<ICoreWebView2> GetComICoreWebView2FromCoreWebView2(
+    winrt::Microsoft::Web::WebView2::Core::CoreWebView2 coreWebView2WinRT)
+{
+    // Get the COM interop interface from the WinRT CoreWebView2.
+    auto interop = coreWebView2WinRT.as<ICoreWebView2Interop2>();
+
+    // Get the COM ICoreWebView2 object from the COM interop interface.
+    winrt::com_ptr<ICoreWebView2> coreWebView2Com;
+    winrt::check_hresult(interop->GetComICoreWebView2(coreWebView2Com.put()));
+	
+	return coreWebView2Com;
+}
+```
+
+## COM to WinRT
+
+```c++
+winrt::Microsoft::Web::WebView2::Core::CoreWebView2 CreateCoreWebView2FromComICoreWebView2(
+    winrt::com_ptr<ICoreWebView2> coreWebView2Com)
+{
+    auto factory = winrt::get_activation_factory<
+        winrt::Microsoft::Web::WebView2::Core::CoreWebView2>();
```

Comment:
Since CoreWebView2 has no constructors or static methods, can I use the activation factory like this? C++/WinRT doesn't generate something for me to implement. Do I need to specify an empty static interface for CoreWebView2 in the MIDL3? Or I can move this to the activation factory of the CoreWebView2Environment?

Reply:
Yikes, I totally overlooked that wrinkle. You can do this in code-behind, no need to pollute the IDL.

Implement a factory manually:
```cpp
namespace winrt::Microsoft::Web::WebView2::Core::factory_implementation
{
    struct CorewebView2 : implements<IActivationFactory, ICoreWebView2Interop>
    {
        [[noreturn]] IInspectable ActivateInstance() const { throw hresult_not_implemented(); }
        // ICoreWebView2Interop methods go here
    };
}
```

and pick it off in your DllGetActivationFactory before delegating to the C++/WinRT generated one.
```cpp
HRESULT WINAPI DllGetActivationFactory(HSTRING classId, IActivationFactory ** factory) noexcept try
{
    auto const& name = (winrt::hstring const&)classId;
    if (name == L"Microsoft.Web.WebView2.CoreCoreWebView2") {
        *factory = winrt::detach_abi(winrt::make<winrt::Microsoft::Web::WebView2::Core::factory_implementation::CoreWebView2>());
        return S_OK;
    } else {
     return WINRT_GetActivationFactory(classId, (void**)factory);
}
catch (...) { return to_hresult(): }
```

Reply:
Right, that makes sense, thanks!